### ISSUE TITLE
Add external source maps to js.Build

### ISSUE THREAD

**Issue by richtera** [gohugoio/hugo#8132]
The current symbol resolver seems to work very well and I have not seen any problems using jsconfig files anymore in the last version. So I think things are working very well, but there are two areas I had fixed in my PR which I would love to create a new PR for. These are external map files and better error reporting.
I was trying to extract these from my jsconfig/tsconfig PR, but I think a lot of things have changed around it. I will put together a new PR for it, but I don't quite understand how a file and it's symbol file is piped through the fingerprint pipeline. I know that this happens properly in sass for example, but I couldn't quite figure out where to look for the code. Also the error reporting of JS doesn't seem to output all of the error information returned from esbuild and will just output the error token without any line or file context. It also skips some kind of import stack to be able to find and repair the error.

**Comment by bep** [comment]
>but I don't quite understand how a file and it's symbol file is piped through the fingerprint pipeline. 

What do you mean by this? What's a "symbol file"?

**Comment by bep** [comment]
OK, I assume you mean "source maps". Yes, I reverted the `external` source map option that was added. If that is needed, there is example code in, e.g. the `dartsass` package.

**Comment by richtera** [comment]
Perfect. That's what I needed. Yes. I will recover the "external" source map support from my PR and the output of ESBuild error messages (currently it seems to only show the main error and no context information to find it unless this was fixed after 79.1) Thanks for the info I will make the PR associated with this issue number.

**Comment by bep** [comment]
https://github.com/gohugoio/hugo/blob/master/resources/resource_transformers/tocss/dartsass/transform.go#L143

**Comment by richtera** [comment]
In order to support es5 targets we might need to support sourceMaps for babel as well and use babel inside of the esbuild pipeline.

**Comment by richtera** [comment]
There is a little bit of a rabbit hole here.

1. I was able to get source maps working for babel
2. It was able to read the inline source maps from esbuild
3. But the output source maps work for some files and not others. I think it's related to requiring larger node dependencies within a particular file, then it will lose context of lines and character offsets within that file.
4. I tried to add a rollup pipeline which also works, but with my current javascript project, rollup is complicated due to the polyfill and resolve configuration. If someone has a good rollup configuration it should work as long as it has `input: "-"` as the input config in the rollup.config.js file.

I have a WIP commit that's mostly working, but I needed to test it on a netlify build. With previous versions I just committed a binary hugo into the repo and then used it to build, but with this particular hugo version I get a
```sh
./build.sh: line 65: /opt/build/repo/platform-themes/local-hugo/hugo: No such file or directory
```
It maybe related to the version of go I am using on my build system vs the netlify container. Is there a "standard" way to create universal test binaries?
